### PR TITLE
[WIP] basic draft of live_region_support

### PR DESCRIPTION
**Pull request by jonahwilliams**
Implements live region for iOS by tracking the contents of live region enabled nodes and performing an announcement if the value changes.

Fixes https://github.com/flutter/flutter/issues/41427

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 1]
```diff
@@ -633,6 +634,16 @@ - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
       object.accessibilityCustomActions = accessibilityCustomActions;
     }
 
+    if (node.HasFlag(flutter::SemanticsFlags::kIsLiveRegion)) {
+      NSNumber* identifer = @(node.id);
+      NSString* label = @(node.label.data());
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: I wonder if we also need to do this for value?

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 2]
```diff
@@ -633,6 +634,16 @@ - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
       object.accessibilityCustomActions = accessibilityCustomActions;
     }
 
+    if (node.HasFlag(flutter::SemanticsFlags::kIsLiveRegion)) {
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Should we remove the mapping of kIsLiveRegion to UpdatesFrequently? I believe giving it that trait doesn't really make sense for us as we currently always fire layoutchanged notifications?

Comment (jonahwilliams) [reply shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Yes, removed

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm (hunk 1 above)
Comment (jonahwilliams) [reply shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: I'm not sure - label/value are fairly distinct on iOS right - do you think we should concatenate them like we do for Android?

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 3]
```diff
@@ -694,6 +706,9 @@ - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
     // TODO(tvolkert): provide meaningful string (e.g. "page 2 of 5")
     UIAccessibilityPostNotification(UIAccessibilityPageScrolledNotification, @"");
   }
+  if (liveRegionUpdated) {
```
Comment (jonahwilliams) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Move to the end to interrupt other announcement

**Comment by jonahwilliams** [comment]
Closing as this doesn't work

**Pull request closed without merging**